Add support for CMSIS-DAP debuggers

ide opened the pull request:
OpenOCD 0.11.0 and up works with CMSIS-DAP debuggers like the Raspberry Pi Debug Probe/Pico Probe and the Flipper Zero.

The Flipper Zero has an installable DAP Link app (through the companion mobile app) that supports SWD. The pinouts are straightforward: the DAP Link app tells you which GPIO pins are SWDCLK and SWDIO, respectively, and of course GND is GND.

Tested all five steps with OpenOCD 0.12.0 on macOS and Flipper Zero firmware 0.88.0. An example step looks like: `./2_backup_flash.sh cmsis-dap mario`. The G&W plays fine after all steps. Also tested with a Picoprobe.

(Also fixed two typos in the main README that are unrelated to this PR but very small.)

ide committed:
Add support for CMSIS-DAP debuggers

OpenOCD 0.11.0 and up works with CMSIS-DAP debuggers like the Raspberry Pi Debug Probe/Pico Probe and the Flipper Zero.

The Flipper Zero has an installable DAP Link app (through the companion mobile app) that supports SWD. The pinouts are straightforward: the DAP Link app tells you which GPIO pins are SWDCLK and SWDIO, respectively, and of course GND is GND.

Tested all five steps with OpenOCD 0.12.0 on macOS and Flipper Zero firmware 0.88.0. An example step looks like: `./2_backup_flash.sh cmsis-dap mario`. The G&W plays fine after all steps. Also tested with a Picoprobe.

(Also fixed two typos in the main README that are unrelated to this PR but very small.)

kbeckmann commented:
Hi and thank you for your PR! Sorry for the slow response. 

#35 was also submitted using cmsis-dap. Maybe the best way forward is to rename the variant to "cmsis-dap" instead of "flipperzero", and add a list of verified debuggers that are known to work (picoprobe, flipper zero etc). What do you think?

ide commented:
@kbeckmann That makes sense and is consistent with the Retro-Go repo. I renamed "flipperzero" to "cmsis-dap" and added some instructions for the Raspberry Pi Debug Probe and Picoprobe. (I don't own a Debug Probe so I looked at the schematic to determine which colored wire corresponds to which SWD pin. Maybe someone with a Debug Probe can verify this.)

kbeckmann commented:
Thank you for the quick change, this is perfect!
